### DISCUSSION TITLE
patterns/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#51] [Announcements]
# patterns/

Abstract Document Abstract Factory Active Object Acyclic Visitor Adapter Aggregator Microservices Ambassador API Gateway Arrange/Act/Assert Async Method Invocation Balking Bridge Builder Business Delegate Bytecode Caching Callback Chain of responsibility Circuit Breaker Client Session Pattern Collecting Parameter Collection Pipeline Combinator Command Commander Component Composite Composite Entity Composite View Context object Converter CQRS Currying Data Access Object Data Bus Data Locality Data Mapper Data Transfer Object Decorator Delegation Dependency Injection Dirty Flag Domain Model Double Buffer Double Checked Locking Double Dispatch Embedded Value Event Aggregator Event-based Asynchronous Event Driven Architecture Event Queue Event Sourcing Execute Around Extension objects Facade Factory Factory Kit Factory Method Fan-Out/Fan-In Feature Toggle Filterer Fluent Interface Flux Flyweight Front Controller Game Loop Guarded Suspension Half-Sync/Half-Async Hexagonal Architecture Identity Map Intercepting Filter Interpreter Iterator Layers Lazy Loading Leader Election Leader/Followers Lockable Object Marker Interface Master-Worker Mediator Memento Metadata Mapping Model-View-Controller Model-View-Presenter Model-View-ViewModel Module Monad Monitor MonoState Multiton Mute Idiom Naked Objects Null Object Object Mother Object Pool Observer Page Controller Page Object Parameter Object Partial Response Pipeline Poison Pill Presentation Model Priority Queue Pattern Private Class Data Producer Consumer Promise Property Prototype Proxy Queue based load leveling Reactor Reader Writer Lock Registry Repository Resource Acquisition Is Initialization Retry Role Object Saga Separated Interface Serialized Entity Pattern Servant Service Layer Service Locator Service to Worker Sharding Singleton Spatial Partition Special Case Specification State Step Builder Strangler Strategy Subclass Sandbox Table Module Template method Thread Pool Throttling Thread Local Storage Tolerant Reader Trampoline Transaction Script Twin Type-Object Unit Of Work Update Method Value Object Version Number Visitor

https://java-design-patterns.com/patterns/

**Comment by DuyTC1811** [comment]
a precious treasure

**Reply by saman488** [reply]
gold mine

**Comment by rajshreeee** [comment]
How did I not find this before

**Comment by ivandimitrov8080** [comment]
I always use this as a reference. Good job.

**Comment by SirOphis** [comment]
this is a treasure

**Comment by OnkarParmar** [comment]
One stop shop!

**Comment by Just0-1** [comment]
so many patterns i don't know

**Comment by javamagiciancom** [comment]
These are some great articles. I'll use them as reference in our next publication about [Java design patterns](https://javamagician.com/)

**Comment by BriqueDeckard** [comment]
OMG. Take that, GoF !

**Comment by cnxutu** [comment]
learning right now, seems useful

**Comment by KaziRifatMorshed** [comment]
Amazing
